src/ngx_http_lua_directive.c

```diff
@@ -487,19 +490,22 @@ ngx_http_lua_rewrite_by_lua(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
             return NGX_CONF_ERROR;
         }
 
+        chunkname_len = strlen((char *) chunkname);
```

Comment:
The use of `strlen` is unfortunate (even if we have to, we should use `ngx_strlen()` instead). Can we make `ngx_http_lua_gen_chunk_name` simply return the string length directly? The string length was readily there and was simply discarded.

Reply:
Good point. Maybe we could change it to:
```
static size_t
ngx_http_lua_gen_chunk_name(ngx_conf_t *cf, const char *tag, size_t tag_len, u_char *chunkname)
```

Reply:
@spacewander `chunkname` pointer is allocated inside this C function, so your prototype cannot return a new pointer but only use a caller-supplied buffer, which does not look right to me (since in that case, you also need to pass in the caller buffer size as an argument).

The following makes more sense to me:

```C
static u_char *
ngx_http_lua_gen_chunk_name(ngx_conf_t *cf, const char *tag, size_t tag_len,
size_t *chunkname_len);
```

It's similar to the standard Lua C API function, `lua_tolstring()`:

```C
const char *lua_tolstring (lua_State *L, int index, size_t *len);
```